src/common/security/robot/context.go

```diff
@@ -78,13 +83,30 @@ func (s *SecurityContext) IsSolutionUser() bool {
 func (s *SecurityContext) Can(ctx context.Context, action types.Action, resource types.Resource) bool {
 	s.once.Do(func() {
 		if s.isSystemLevel {
-			s.evaluator = rbac.NewProjectEvaluator(s.ctl, rbac.NewBuilderForPolicies(s.GetUsername(), s.policy))
+			var proPolicies []*types.Policy
+			var sysPolicies []*types.Policy
+			for _, p := range s.policy {
+				if strings.HasPrefix(p.Resource.String(), robot.SCOPESYSTEM) {
+					sysPolicies = append(sysPolicies, p)
+				} else if strings.HasPrefix(p.Resource.String(), robot.SCOPEPROJECT) {
+					proPolicies = append(proPolicies, p)
+				}
+			}
+			if len(sysPolicies) != 0 {
+				s.sysEvaluator = system.NewEvaluator(s.GetUsername(), sysPolicies)
+			} else if len(proPolicies) != 0 {
+				s.proEvaluator = rbac.NewProjectEvaluator(s.ctl, rbac.NewBuilderForPolicies(s.GetUsername(), proPolicies))
+			}
+
 		} else {
-			s.evaluator = rbac.NewProjectEvaluator(s.ctl, rbac.NewBuilderForPolicies(s.GetUsername(), s.policy, filterRobotPolicies))
+			s.proEvaluator = rbac.NewProjectEvaluator(s.ctl, rbac.NewBuilderForPolicies(s.GetUsername(), s.policy, filterRobotPolicies))
 		}
 	})
 
-	return s.evaluator != nil && s.evaluator.HasPermission(ctx, resource, action)
+	if s.isSystemLevel {
+		return (s.sysEvaluator != nil && s.sysEvaluator.HasPermission(ctx, resource, action)) || (s.proEvaluator != nil && s.proEvaluator.HasPermission(ctx, resource, action))
+	}
+	return s.proEvaluator != nil && s.proEvaluator.HasPermission(ctx, resource, action)
```

Comment:
```golang
if s.isSystemLevel {
    var evaluators evaluator.Evaluators
    if len(sysPolicies) > 0 {
        evaluators = evaluators.Add(system.NewEvaluator(s.GetUsername(), sysPolicies))
    }
    if len(proPolicies) > 0 {
        evaluators = evaluators.Add(rbac.NewProjectEvaluator(s.ctl, rbac.NewBuilderForPolicies(s.GetUsername(), proPolicies)))
    }

    s.evaluator = evaluators

    // or check the len of evaluators
    // if len(evaluators) > 1 {
    //   s.evaluator = evaluators
    // } else if len(evaluators) == 1 {
    //  s.evaluator = evaluators[0]    
    // }
} else {
    s.evaluator = rbac.NewProjectEvaluator(s.ctl, rbac.NewBuilderForPolicies(s.GetUsername(), s.policy, filterRobotPolicies))
} 
```

Reply:
will use evaluator.Evaluators for instead